lib/config.js

```diff
@@ -132,23 +134,28 @@ class Config {
         });
     }
 
+    getPluginRootPath() {
+        return this.options.cwd;
+    }
+
     /**
-     * Loads the config options from a config specified on the command line.
+     * Loads the config options from a config specified on the command line, relative to the CWD.
      * @param {string} [config] A shareable named config or path to a config file.
      * @returns {void}
      */
     loadSpecificConfig(config) {
         if (config) {
             debug(`Using command line config ${config}`);
-            const isNamedConfig =
-                isResolvable(config) ||
-                isResolvable(`eslint-config-${config}`) ||
-                config.charAt(0) === "@";
-
-            this.specificConfig = ConfigFile.load(
-                isNamedConfig ? config : path.resolve(this.options.cwd, config),
-                this
-            );
+            const resolvedPath = path.resolve(this.options.cwd, config);
+
+            this.specificConfigInfo = {
+                filePath: resolvedPath,
+                config: ConfigFile.load(
+                    resolvedPath,
```

Comment:
Does this support shareable configs as is?

Reply:
No, this doesn't work on `master` anyway because the cache key is incorrect when loading the config:

```bash
$ echo 'foo' > test.js
$ eslint --no-eslintrc --config eslint-config-eslint --no-ignore test.js
(no errors)
$ eslint --no-eslintrc --config packages/eslint-config-eslint/index.js --no-ignore test.js

/Users/tkatz/code/eslint/test.js
  1:1  error  Use the global form of 'use strict'                                    strict
  1:1  error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
  1:1  error  'foo' is not defined                                                   no-undef
  1:4  error  Missing semicolon                                                      semi

✖ 4 problems (4 errors, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

It seems undesirable to change behavior based on whether a shareable config is resolvable, so I removed the logic since it wasn't working anyway.

Reply:
I got it.

It looks like a regression (that exists while a long time?), but nobody reported it. I'm OK to remove.

Reply:
Ah... but it's documented: 

> It also accepts a module ID of a sharable config.
> https://eslint.org/docs/user-guide/command-line-interface#-c---config

Reply:
I "fixed" the issue in https://github.com/eslint/eslint/pull/11388/commits/f8459c613f6ef5918bdecf89b1098dbf2ab767f5 by removing that line from the documentation. This has been broken since v4.1.0 and apparently no one has noticed, so it seems like it's probably not worth adding complexity to make it work again.